src/main/java/com/appium/capabilities/Capabilities.java

```diff
@@ -1,36 +1,62 @@
 package com.appium.capabilities;
 
-import static com.appium.utils.ConfigFileManager.CAPS;
-
+import com.appium.device.AtdEnvironment;
+import com.appium.manager.RemoteAppiumManager;
+import com.appium.utils.FigletHelper;
 import com.appium.utils.JsonParser;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
 import org.json.JSONArray;
 import org.json.JSONObject;
+import org.json.JSONTokener;
 
+import java.io.InputStream;
+import java.net.ConnectException;
+import java.net.InetAddress;
 import java.nio.file.FileSystems;
 import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.logging.Logger;
+
+import static com.appium.utils.ConfigFileManager.CAPS;
+
+public class Capabilities {
```

Comment:
Not convinced in calling as Capabilities.

Reply:
How about CapabilityConfiguration (which is responsible for the parsing, validation and inference)?

The idea is to move away from the Manager, Validator and other classes that end with "er", "or", etc. The reason being these classes lead to data being in Plain Old Java Objects and their related functionality in other places which breaks encapsulation. The code will be a lot more intuitive if we move to Job Description based names (Example: HostMachine, Device, Platform etc.) for classes which have data and related functionality together.

Reply:
Do suggest any alternatives. I am happy to go with any other name which is not manager, service etc. which conveys intent. At the moment Capabilities is also having too many responsibilities, so it may be hard to name it. We can revisit the name once we reduce it to a single responsibility.

Reply:
I suggest revisiting it later.